AttentionClipperが新ロボットモデルで動作しない

#1320 の原因を調べたところ、jsk_2015_05_baxter_apcのrecognition_in_hand.launchの中に、
```
  <node name="in_hand_clipper"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/AttentionClipper $(arg MANAGER)"
        clear_params="true">
    <remap from="~input/points" to="$(arg INPUT_CLOUD)" />
    <rosparam>
      use_multiple_attention: true
      initial_pos_list: [[0.0, -0.1, 0.3], [0.0, -0.1, 0.3]]
      initial_rot_list: [[0, 0, 0], [0, 0, 0]] 
      dimensions: [[0.3, 0.2, 0.3], [0.3, 0.2, 0.3]]
      frame_id_list: [left_custom_vacuum_link_base, right_custom_vacuum_link_base]
      prefixes: [left_hand, right_hand]
    </rosparam>
  </node>
```
とあり、frame_id_listの中に入ってるリンクのうち、right_custom_vacuum_link_baseが新ロボットモデルに入っていないのが原因であるように思えた。しかし、認識プログラムについての知識が不足しており、わからない。
私のmilestoneではしばらく気にならないので、 @knorth55 にお願いしたい。

```
$ roslaunch jsk_2015_05_baxter_apc setup_torso.launch
```
を起動しているターミナルに、いつの間にか
```
[ERROR] [1461662792.586700906]: [virtual void jsk_pcl_ros::AttentionClipper::clipPointcloud(const ConstPtr&)] Transform error: Failed to lookup transformation from kinect2_torso_rgb_optical_frame to right_custom_vacuum_link_base
```
というエラーが出るようになっているので、すくなくともそこでエラーが起きています。

--- a/jsk_2016_01_baxter_apc/launch/recognition_in_hand.launch
+++ b/jsk_2016_01_baxter_apc/launch/recognition_in_hand.launch
@@ -1,0 +1,152 @@
+<launch>
+
+  <arg name="INPUT_CLOUD" />
+  <arg name="INPUT_IMAGE" />
+  <arg name="MANAGER" />
+
+
+  <node name="in_hand_clipper"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/AttentionClipper $(arg MANAGER)"
+        clear_params="true">
+    <remap from="~input/points" to="$(arg INPUT_CLOUD)" />
+    <rosparam>
+      use_multiple_attention: true
+      initial_pos_list: [[0.0, -0.1, 0.3], [0.0, 0.0, 0.15]]
+      initial_rot_list: [[0, 0, 0], [0, 0, 0]]
+      dimensions: [[0.3, 0.2, 0.3], [0.3, 0.3, 0.2]]
+      frame_id_list: [left_custom_vacuum_link_base, right_gripper_vacuum_pad]
+      prefixes: [left_hand, right_hand]
+    </rosparam>
+  </node>
+
+
+  <!-- debug info -->
+  <node name="extract_indices_left_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/ExtractIndices $(arg MANAGER)">
+    <remap from="~input" to="$(arg INPUT_CLOUD)" />
+    <remap from="~indices" to="in_hand_clipper/left_hand/point_indices" />
+    <remap from="~output" to="~points" />
+    <rosparam>
+      keep_organized: true
+    </rosparam>
+  </node>
+  <node name="extract_indices_right_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/ExtractIndices $(arg MANAGER)">
+    <remap from="~input" to="$(arg INPUT_CLOUD)" />
+    <remap from="~indices" to="in_hand_clipper/right_hand/point_indices" />
+    <remap from="~output" to="~points" />
+    <rosparam>
+      keep_organized: true
+    </rosparam>
+  </node>
+
+
+  <!-- self filter -->
+  <include file="$(find jsk_2015_05_baxter_apc)/launch/include/self_filter.launch">
+    <arg name="INPUT_CLOUD" value="extract_indices_left_hand/points" />
+    <arg name="OUTPUT_CLOUD" value="extract_indices_left_hand/points_filtered" />
+  </include>
+  <include file="$(find jsk_2015_05_baxter_apc)/launch/include/self_filter.launch">
+    <arg name="INPUT_CLOUD" value="extract_indices_right_hand/points" />
+    <arg name="OUTPUT_CLOUD" value="extract_indices_right_hand/points_filtered" />
+  </include>
+
+
+  <node name="pc_to_pi_left_hand"
+        pkg="nodelet" type="nodelet"
+        args="standalone jsk_pcl/OrganizedPointCloudToPointIndices $(arg MANAGER)">
+    <remap from="~input" to="extract_indices_left_hand/points_filtered" />
+  </node>
+  <node name="pc_to_pi_right_hand"
+        pkg="nodelet" type="nodelet"
+        args="standalone jsk_pcl/OrganizedPointCloudToPointIndices $(arg MANAGER)">
+    <remap from="~input" to="extract_indices_right_hand/points_filtered" />
+  </node>
+
+
+  <!-- object segmented image -->
+  <!-- convert point indices to mask image -->
+  <node name="pi_to_mask_left_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/PointIndicesToMaskImage $(arg MANAGER)">
+    <remap from="~input" to="pc_to_pi_left_hand/output" />
+    <remap from="~input/image" to="$(arg INPUT_IMAGE)" />
+    <rosparam>
+      queue_size: 1000
+    </rosparam>
+  </node>
+  <node name="pi_to_mask_right_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_pcl/PointIndicesToMaskImage $(arg MANAGER)">
+    <remap from="~input" to="pc_to_pi_right_hand/output" />
+    <remap from="~input/image" to="$(arg INPUT_IMAGE)" />
+    <rosparam>
+      queue_size: 1000
+    </rosparam>
+  </node>
+  <!-- process mask image -->
+  <node name="closing_left_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_perception/Closing $(arg MANAGER)">
+    <remap from="~input" to="pi_to_mask_left_hand/output" />
+    <rosparam>
+      size: 8
+      iterations: 4
+    </rosparam>
+  </node>
+  <node name="closing_right_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_perception/Closing $(arg MANAGER)">
+    <remap from="~input" to="pi_to_mask_right_hand/output" />
+    <rosparam>
+      size: 8
+      iterations: 4
+    </rosparam>
+  </node>
+  <!-- apply mask image -->
+  <node name="apply_mask_left_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_perception/ApplyMaskImage $(arg MANAGER)">
+    <remap from="~input" to="$(arg INPUT_IMAGE)" />
+    <remap from="~input/mask" to="closing_left_hand/output" />
+    <rosparam>
+      queue_size: 1000
+    </rosparam>
+  </node>
+  <node name="mask_to_label_left_hand"
+        pkg="jsk_perception" type="mask_image_to_label.py">
+    <remap from="~input" to="apply_mask_left_hand/output/mask" />
+  </node>
+  <node name="apply_mask_right_hand"
+        pkg="nodelet" type="nodelet"
+        args="load jsk_perception/ApplyMaskImage $(arg MANAGER)">
+    <remap from="~input" to="$(arg INPUT_IMAGE)" />
+    <remap from="~input/mask" to="closing_right_hand/output" />
+    <rosparam>
+      queue_size: 1000
+    </rosparam>
+  </node>
+  <node name="mask_to_label_right_hand"
+        pkg="jsk_perception" type="mask_image_to_label.py">
+    <remap from="~input" to="apply_mask_right_hand/output/mask" />
+  </node>
+
+  <!-- object recognition -->
+  <group ns="left_hand">
+    <include file="$(find jsk_2015_05_baxter_apc)/launch/include/object_recognition.launch">
+      <arg name="INPUT_IMAGE" value="/apply_mask_left_hand/output" />
+      <arg name="INPUT_LABEL" value="/mask_to_label_left_hand/output" />
+    </include>
+  </group>
+  <group ns="right_hand">
+    <include file="$(find jsk_2015_05_baxter_apc)/launch/include/object_recognition.launch">
+      <arg name="INPUT_IMAGE" value="/apply_mask_right_hand/output" />
+      <arg name="INPUT_LABEL" value="/mask_to_label_right_hand/output" />
+    </include>
+  </group>
+
+
+</launch>

--- a/jsk_2016_01_baxter_apc/launch/setup_torso.launch
+++ b/jsk_2016_01_baxter_apc/launch/setup_torso.launch
@@ -1,0 +1,26 @@
+<launch>
+
+
+  <include file="$(find jsk_2015_05_baxter_apc)/launch/include/kinect2_torso.launch" />
+
+
+  <arg name="MANAGER" value="doura_manager" />
+  <node name="$(arg MANAGER)"
+        pkg="nodelet" type="nodelet" args="manager" />
+
+
+  <include file="$(find jsk_2015_05_baxter_apc)/launch/segmentation_in_bin_gtol.launch">
+    <arg name="INPUT_IMAGE" value="/kinect2_torso/hd/image_color" />
+    <arg name="INPUT_DEPTH" value="/kinect2_torso/hd/image_depth_rect" />
+    <arg name="INPUT_CLOUD" value="/kinect2_torso/hd/points" />
+    <arg name="MANAGER" value="$(arg MANAGER)" />
+  </include>
+
+  <include file="$(find jsk_2016_01_baxter_apc)/launch/recognition_in_hand.launch">
+    <arg name="INPUT_IMAGE" value="/kinect2_torso/hd/image_color" />
+    <arg name="INPUT_CLOUD" value="/kinect2_torso/hd/points" />
+    <arg name="MANAGER" value="$(arg MANAGER)" />
+  </include>
+
+
+</launch>

